parser/statement_parsing.go

```diff
@@ -8,7 +8,7 @@ import (
 
 func (p *Parser) parseStatement() ast.Statement {
 	switch p.curToken.Type {
-	case token.InstanceVariable, token.Ident, token.Constant:
+	case token.InstanceVariable, token.Ident, token.Constant, token.Class, token.Module:
```

Comment:
Since we already have class token and module token, we should move them into their own cases like:
```go
switch p.curToken.Type {
case token.Class:
  p.parseClassStatement()
case token.Module:
  p.parseModuleStatement()
....
}